contracts/core/merge/MergeFactory.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
+
+import "./IMergeV2.sol";
+import {GenericMerge} from "./GenericMerge.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MergeFactory is Ownable {
+    address public constant wewe = 0x6b9bb36519538e0C073894E964E90172E1c0B41F;
+    address[] public tokens;
+    // token => merge
+    mapping(address => address) public merges;
+    mapping(address => bool) public allowedDeployers;
+
+    constructor() {
+        allowedDeployers[msg.sender] = true;
+    }
+
+    function getMergeCount() external view returns (uint256) {
+        return tokens.length;
+    }
+
+    function createMerge(address token, uint256 rate) external onlyDeployer returns (address) {
+        require(merges[token] == address(0), "MergeFactory: Merge already exists");
+        address merge = address(new GenericMerge(token, wewe));
+        IMergeV2(merge).setRate(rate);
+        _setMerge(token, merge);
+        return merge;
```

Comment:
Could be nice to create an script to verify merge contracts deployed from factory, because without the script we will need to do it manually

Reply:
Yeah, sounds good.  Basescan should find it as "similar" too.